### ISSUE TITLE
A collection of tests failing while testing during debian package build

### ISSUE THREAD

**Issue by yarikoptic** [fail2ban/fail2ban#2024]
yet to look in detail and most likely "my fault" one way or another. Just wanted to report first in case @sebres has an immediate clue. Cheers

```shell
..
cd build && LC_ALL=C.UTF-8 FAIL2BAN_CONFIG_DIR="/home/yoh/deb/gits/fail2ban/config" PYTHONPATH="/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/" scripts-*/fail2ban-testcases --no-network
......................s.........................................................................sssss.........................s.................................ss....................................................F...sss.............s......sssssssssssssss...............................................................................................EFail2ban 0.10.2 test suite. Python 3.6.4 (default, Jan  5 2018, 02:13:53) [GCC 7.2.0]. Please wait...
........
..Exception in thread _TestCaseWorker:  
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2banserver.py", line 186, in start
    raise ServerExecutionException('Async configuration of server failed')
fail2ban.client.fail2bancmdline.ServerExecutionException: Async configuration of server failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 410, in _testStartForeground
    self.execCmd(SUCCESS, ("-f",) + startparams, "start")
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 401, in execCmd
    (self.exec_command_line[1:] + startparams + args))
  File "/usr/lib/python3.6/unittest/case.py", line 733, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/lib/python3.6/unittest/case.py", line 178, in handle
    callable_obj(*args, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2banserver.py", line 235, in exec_command_line
    if server.start(argv):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2banserver.py", line 223, in start
    exit(-1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2bancmdline.py", line 321, in exit
    Fail2banCmdLine._exit(code)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 371, in _test_exit
    raise FailExitException()
fail2ban.tests.fail2banclienttestcase.FailExitException

F....................E.............ssssss......................E.E..........
======================================================================
ERROR: testSampleRegexsZZZ-SSHD-OBSOLETE-MULTILINE (fail2ban.tests.samplestestcase.FilterSamplesRegex)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/samplestestcase.py", line 169, in testFilter
    flt = self._readFilter(fltName, name, basedir, opts=opts)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/samplestestcase.py", line 102, in _readFilter
    filterConf.getOptions({})
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 319, in getOptions
    self, "Definition", self._configOpts, pOpts)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 147, in getOptions
    return self._cfg.getOptions(section, *args, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 245, in getOptions
    v = self.get(sec, optname, vars=pOptions)
  File "/usr/lib/python3.6/configparser.py", line 800, in get
    d)
  File "/usr/lib/python3.6/configparser.py", line 394, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 437, in _interpolate_some
    section, map, depth + 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 437, in _interpolate_some
    section, map, depth + 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 434, in _interpolate_some
    option, section, rawval, var) from None
configparser.InterpolationMissingOptionError: Bad value substitution: option 'failregex' in section 'Definition' contains an interpolation key '__prefix_line' which is not a valid option name. Raw value: '%(cmnfailre)s\n<mdre-<mode>>'

======================================================================
ERROR: testMultilineSshd (fail2ban.tests.fail2banregextestcase.Fail2banRegexTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banregextestcase.py", line 243, in testMultilineSshd
    self.assertTrue(fail2banRegex.start(args))
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2banregex.py", line 586, in start
    if not self.readRegex(cmd_regex, 'fail'): # pragma: no cover
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/fail2banregex.py", line 339, in readRegex
    reader.getOptions(None)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 319, in getOptions
    self, "Definition", self._configOpts, pOpts)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 147, in getOptions
    return self._cfg.getOptions(section, *args, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configreader.py", line 245, in getOptions
    v = self.get(sec, optname, vars=pOptions)
  File "/usr/lib/python3.6/configparser.py", line 800, in get
    d)
  File "/usr/lib/python3.6/configparser.py", line 394, in before_get
    self._interpolate_some(parser, option, L, value, section, defaults, 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 437, in _interpolate_some
    section, map, depth + 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 437, in _interpolate_some
    section, map, depth + 1)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/client/configparserinc.py", line 62, in _interpolate_some
    parser, option, accum, rest, section, map, *args, **kwargs)
  File "/usr/lib/python3.6/configparser.py", line 434, in _interpolate_some
    option, section, rawval, var) from None
configparser.InterpolationMissingOptionError: Bad value substitution: option 'failregex' in section 'Definition' contains an interpolation key '__prefix_line' which is not a valid option name. Raw value: '%(cmnfailre)s\n<mdre-<mode>>'

======================================================================
ERROR: testJournalFilesArg (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
IndexError: list index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/filtertestcase.py", line 1185, in testJournalFilesArg
    self._initFilter(journalfiles=jrnlfile)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/filtertestcase.py", line 1158, in _initFilter
    self.filter = Filter_(self.jail, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/server/filtersystemd.py", line 62, in __init__
    self.__journal = journal.Reader(**jrnlargs)
  File "/usr/lib/python3/dist-packages/systemd/journal.py", line 174, in __init__
    super(Reader, self).__init__(flags, path, files)
SystemError: <method-wrapper '__init__' of Reader object at 0x7f9c5b074888> returned a result with an error set

======================================================================
ERROR: testJournalPathArg (fail2ban.tests.filtertestcase.MonitorJournalFailures<FilterSystemd>(monitorjournalfailures_FilterSystemd))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/filtertestcase.py", line 1191, in testJournalPathArg
    self._initFilter(journalpath=jrnlpath)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/filtertestcase.py", line 1158, in _initFilter
    self.filter = Filter_(self.jail, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/server/filtersystemd.py", line 62, in __init__
    self.__journal = journal.Reader(**jrnlargs)
  File "/usr/lib/python3/dist-packages/systemd/journal.py", line 174, in __init__
    super(Reader, self).__init__(flags, path, files)
FileNotFoundError: [Errno 2] No such file or directory

======================================================================
FAIL: testUnaccessibleLogFile (fail2ban.tests.filtertestcase.LogFileMonitor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/filtertestcase.py", line 681, in testUnaccessibleLogFile
    self.assertTrue(failure_was_logged != is_root)
AssertionError: False is not true

======================================================================
FAIL: testServerActions_NginxBlockMap (fail2ban.tests.fail2banclienttestcase.Fail2banServerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 323, in wrapper
    self._wait_for_srv(tmp, True, startparams=startparams)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 381, in _wait_for_srv
    % (startparams,)
Exception: Unexpected: Socket file does not exists.
Start failed: ('-c', '/home/yoh/.tmp/f2b-tempz4143lay/config', '-s', '/home/yoh/.tmp/f2b-tempz4143lay/f2b.sock', '-p', '/home/yoh/.tmp/f2b-tempz4143lay/f2b.pid', '--logtarget', 'INHERITED', '--loglevel', 'INFO', '--syslogsocket', 'auto', '--timeout', '29')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/utils.py", line 206, in wrapper
    return f(self, tmp, *args, **kwargs)
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 333, in wrapper
    self.stopAndWaitForServerEnd()
  File "/home/yoh/deb/gits/fail2ban/.pybuild/pythonX.Y_3.6/build/fail2ban/tests/fail2banclienttestcase.py", line 316, in _stopAndWaitForServerEnd
    self.assertTrue(phase.get('end', None))
AssertionError: None is not true

----------------------------------------------------------------------
Ran 438 tests in 111.067s

FAILED (failures=2, errors=4, skipped=34)
```

**Comment by sebres** [comment]
Can you provide your config `/home/yoh/deb/gits/fail2ban/config` (e. g. as zip-attachment) or some info about minimal pre-requirements for custom config-dir to test fail2ban like you do it.

I assume some cases are just missing skip-if-not-stock and some are just wrong (incompatible to v.0.10), but I don't know what exactly it is in each failed test-case.

I rebased your branch #2025 and make some amendments now (will push it soon as alternative to your's PR).

**Comment by yarikoptic** [comment]
I was (by all means and purpose) testing against our config which is within the repository by pointing to it with `FAIL2BAN_CONFIG_DIR="/home/yoh/deb/gits/fail2ban/config" `, and that one is the one we ship. The only one custom I see now is
```shell
$> git status -- config 
On branch debian
Your branch is ahead of 'origin/debian' by 830 commits.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	config/filter.d/sshd.local_wastroubleshootingsomething
```
which I not removed.

Above failures were due to
- invoking tests outside of the source build tree (well -- under `build/` subdirectory), so code must have no longer relied on having stock configs under `config/` and should have used that env var as we did before
- different location of the journal file on my debian system

All of them AFAIK were addressed in #2025 .  I guess now I will need (later) to compare that state against your rebased one (good that I just picked up those patches and not merged into debian branch... not sure what rebasing them helps here with).  With those fixes in #2025 (besides the manpages) tests pass while building debian packages (yet to rebuild to see if builds fine on older debian/ubuntus)

**Comment by sebres** [comment]
> not sure what rebasing them helps here with

You can rebase it back to your branch (then will get 2 commits of me, if you'll remove d1afbb566f0304487b5d578b4aacef8e647ee74b).

I rebased because of #2026 and because I have deleted d1afbb566f0304487b5d578b4aacef8e647ee74b, because now allowed to specify `use_stock_cfg` also in stock case (None is just default now that corresponds to tuple `('action.d', 'filter.d')`).
